### ISSUE TITLE
Error when inspecting a Hash

### ISSUE THREAD

**Issue by frsantos** [ksss/rubocop-yard#19]
The following code, with a hash option associated to no param, fails

```ruby
class Test
  # @param [Hash{Symbol=>Object}] options The options
  # @option [Integer] :op Option associated to no param
  def test(options)
  end
end
```

with 

```txt
An error occurred while YARD/CollectionStyle cop was inspecting /app/test.rb.                                                                                                                                                                                                                                                                                                                 
undefined method `join' for nil:NilClass                                                                                                                                                                                                                                                                                                                                                      
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/helper.rb:25:in `block in each_types_explainer'                                                                                                                                                                                                                                                                                
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/helper.rb:21:in `each'                                                                                                                                                                                                                                                                                                         
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/helper.rb:21:in `each_types_explainer'                                                                                                                                                                                                                                                                                         
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/collection_style.rb:63:in `check'                                                                                                                                                                                                                                                                                              
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/collection_style.rb:55:in `block in on_new_investigation'                                                                                                                                                                                                                                                                      
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/collection_style.rb:51:in `each'                                                                                                                                                                                                                                                                                               
/usr/local/bundle/gems/rubocop-yard-0.8.1/lib/rubocop/cop/yard/collection_style.rb:51:in `on_new_investigation'
```

**Comment by ksss** [comment]
@frsantos 
Thank you for reporting.
I couldn't reproduce the issue in my environment. Could you please tell me the versions of Ruby, RuboCop, and YARD you are using?

**Comment by ksss** [comment]
@frsantos Thank you for reporting.
I fixed by v0.8.2.
However, The cause of this issue is that when analyzing the `@option` tag, YARD's parser could not retrieve the type information.

Please see `@types=nil`.

```rb
::YARD::DocstringParser.new.parse("@option [Integer] :op Option associated to no param")
#=>
#<YARD::DocstringParser:0x000000011162baf8
 @directives=[],
 @handler=nil,
 @library=#<YARD::Tags::Library:0x0000000111370548 @factory=#<YARD::Tags::DefaultFactory:0x00000001112f85c0>>,
 @object=nil,
 @raw_text="@option [Integer] :op Option associated to no param",
 @reference=nil,
 @state=#<OpenStruct>,
 @tags=
  [#<YARD::Tags::OptionTag:0x000000011166cb20
    @name="[Integer]",
    @pair=#<YARD::Tags::DefaultTag:0x000000011166cb70 @defaults=nil, @name=":op", @tag_name="option", @text="Option associated to no param", @types=nil>,
    @tag_name="option",
    @text=nil,
    @types=nil>],
 @text="">
```
This seems to include issues related to format, so I'm considering whether a new cop could address this.

**Comment by frsantos** [comment]
> @frsantos Thank you for reporting. I fixed by v0.8.2. 

Thank you!

> However, The cause of this issue is that when analyzing the `@option` tag, YARD's parser could not retrieve the type information.
> 
> Please see `@types=nil`.
> 
> ```ruby
> ::YARD::DocstringParser.new.parse("@option [Integer] :op Option associated to no param")
> #=>
> #<YARD::DocstringParser:0x000000011162baf8
>  @directives=[],
>  @handler=nil,
>  @library=#<YARD::Tags::Library:0x0000000111370548 @factory=#<YARD::Tags::DefaultFactory:0x00000001112f85c0>>,
>  @object=nil,
>  @raw_text="@option [Integer] :op Option associated to no param",
>  @reference=nil,
>  @state=#<OpenStruct>,
>  @tags=
>   [#<YARD::Tags::OptionTag:0x000000011166cb20
>     @name="[Integer]",
>     @pair=#<YARD::Tags::DefaultTag:0x000000011166cb70 @defaults=nil, @name=":op", @tag_name="option", @text="Option associated to no param", @types=nil>,
>     @tag_name="option",
>     @text=nil,
>     @types=nil>],
>  @text="">
> ```
> 
> This seems to include issues related to format, so I'm considering whether a new cop could address this.

That would be useful, yes.

**Comment by ksss** [comment]
@frsantos I also release v0.9.0 with new cop `YARD/TagTypePosition`.
Please check it out.
